### ISSUE TITLE
Brand Icons for entities from multiple integrations

### ISSUE THREAD

**Issue by RASBR** [home-assistant/frontend#14620]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

I have some Shelly 1PM (and other models) devices I setup with the same exact steps. Those devices are provided to HA from two integrations Shelly and Unifi Network. The thing is that they are not represented the same in HA especially in regards to Brand Logo!

Attached some screenshots.

### Describe the behavior you expected

Device brand icons to be represented according to their manufacturer.

### Steps to reproduce the issue

1. No specific steps!!
![01 Multiple Integration Entities](https://user-images.githubusercontent.com/63133627/206373788-6083270f-61d7-4e7d-b33c-9c86867d363c.jpg)
![02 Device ONE](https://user-images.githubusercontent.com/63133627/206373791-c6bb7d34-ab80-45f8-be0e-f9a8b83776fa.jpg)
![03 Device TWO](https://user-images.githubusercontent.com/63133627/206373792-c804cbb8-f5ed-434c-9188-915e272dab95.jpg)

### What version of Home Assistant Core has the issue?

Home Assistant 2022.11.0

### In which browser are you experiencing the issue with?

Version 108.0.5359.95 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

HA is installed on Intel NUC

**Comment by steverep** [comment]
This is still relevant

**Comment by steverep** [comment]
Still relevant IMO

**Comment by spacegaier** [comment]
Yes, still relevant. There is an arch discussion that will potentially enhance the backend so we can reliably get the information from there. That would then help the frontend to show the correct info: https://github.com/home-assistant/architecture/discussions/936

**Comment by steverep** [comment]
@spacegaier I do believe that proposal would help the situation, but would not completely solve the icon issue. I'll comment more there.

**Comment by mpoettgen** [comment]
Experiencing the same problem for Shelly devices with a combination of the Shelly integration and a "AVM FRITZ!Box Tools" integration. Expecting to see Shelly icons for the Shelly devices, but some (not all) are represented by FRITZ icons.

**Comment by vdrHorst** [comment]
There seems to be an architectural discussion on auto determining primairy or secondary integration of a device. Hope that can be addressed soon as the different brand icons are at least confusing.
But is there a workaround known for (manual) correction? Should I remove the second integration, so all double registrations fall back to its remaining integration as primary, after which I can reinstall the second integration? My concerned integrations on this topic are primary Shelly devices and secondary Ubiquity network.

**Comment by GitHubGoody** [comment]
+1

All of these should indicate ESPHome as the primary integration. I've even disabled the UniFi device tracker on each of them.
![image](https://github.com/home-assistant/frontend/assets/46235745/66489771-f4dc-49ad-aad6-360df81c8c7d)

**Comment by ozwo71** [comment]
+1
I have the same issue. The icon in the device list should be configurable (set to the "main" integration).
E.g. Sonos/Ubiquiti:
![grafik](https://github.com/home-assistant/frontend/assets/46357623/de237b86-32fd-4d90-b9c4-eb389d9e0efc)
![grafik](https://github.com/home-assistant/frontend/assets/46357623/59877e90-13b4-4e9a-87ae-9f8a429a5fe3)
![grafik](https://github.com/home-assistant/frontend/assets/46357623/d295e219-b754-46bb-9577-157d15ef01df)

Same with ESPHome/Ubiquiti:
![grafik](https://github.com/home-assistant/frontend/assets/46357623/0ff82c27-2e53-4c16-9199-b7df93d23da2)

It seems that the icon in the device list is taken from the first integration that is listed in the device info.

It would be nice to make this configurable (set "main" integration for device).

Thank you

**Comment by ozwo71** [comment]
This makes it clear:
![grafik](https://github.com/home-assistant/frontend/assets/46357623/74cb40be-183d-40fb-ae62-7cd8b4d2fc5b)

The icon is not taken from the "Manufacturer" but from the first entry in the "Integration" column of the device list.
I would expect the icon is defined by manufacturer.

**Comment by dacootmeister** [comment]
I encounter the same with Asusrouter integration, Sonos, Samsung and Homewizard devices
![image](https://github.com/home-assistant/frontend/assets/35705080/c2b2d9a4-d8b5-40d0-ae4c-394db8bd5685)
![image](https://github.com/home-assistant/frontend/assets/35705080/850041df-50e1-4ddc-a2b5-1c8db887fd00)

**Comment by Tomanic-glitch** [comment]
Can confirm this happens to me as well, e.g. the TP-Link plugs sometimes having Nmap icons instead of TP-Link based on which integration is listed first

![image](https://github.com/home-assistant/frontend/assets/61036575/5249461e-5d28-4cbf-8e5a-590dd9e2a829)

**Comment by samedarkclouds** [comment]
Still relevant. Battery notes and Unifi are destroying any sort by integration....

**Comment by ThomasHeinrichSchmidt** [comment]
Same issue, it's confusing, did not find a workaround to swap the integrations. Would like to see **Shelly** icon for all Shellies.
![Shelly devices](https://github.com/home-assistant/frontend/assets/8068047/1d5b7969-e1bb-4788-833c-436b6e77f72b)

**Comment by BDBAfH** [comment]
+1

**Comment by lee-booy** [comment]
Same for me, makes it hard to quickly identify devices when everything has a Unifi icon against it :)

**Comment by Cyruz143** [comment]
I'd also like to see this resolved.

If you wanted a nice, UX friendly solution, you could make the device info panel a list where you can drag the order of the integration around, top one being the primary.

E.g.
![image](https://github.com/home-assistant/frontend/assets/7225500/3c83cb4f-d24b-4750-b470-23c15469dfef)

**Comment by samedarkclouds** [comment]
Same... this really sucks when trying to organize.

**Comment by smartypants1138** [comment]
I would like to see this resolved too, current implementation makes organization difficult and confusing. I propose an option that allows for setting a main vendor for each device.

**Comment by jouster1974** [comment]
so I think I have similar wher e a device shows multiple installations....and Im also seeing when some devices are resatrt/changed, they then affect those other integrations

In the below example, the integration is Solarman...but its also showing as UNifi and Utility meter.  There is an option to Remove device from Unifi (in red text) but it doesn seem to do anything

![image](https://github.com/user-attachments/assets/d1a513cc-28bc-4176-9a15-ff6be6156c5f)

![image](https://github.com/user-attachments/assets/d5bf8fd8-e1b3-4aad-8cb1-e142b7bb36d0)

**Comment by smartypants1138** [comment]
I noticed this on a device page with the wrong icon, this is really sloppy and should be fixed: 
![image](https://github.com/user-attachments/assets/38d815b3-a32b-499a-9383-b811b328d875)

**Comment by amjadsaadeh** [comment]
Similat problem here, I got a Shelly device and it is primarly assognes to the FRITZ!Box Integration instead to the Shelly integration. It's mainly about the wrong icon.

**Comment by ThomasHeinrichSchmidt** [comment]
For me the issue has been resolved since one of the recent updates.
![Shelly icons](https://github.com/user-attachments/assets/acc17318-489a-45aa-bba0-c156be587b4b)

**Comment by RASBR** [comment]
Although a bit late, but thanks... seems much better now.
